src/Test/PHPUnit/AbstractCacheConstraint.php

```diff
@@ -41,17 +43,25 @@ public function __construct($header = null)
 
     /**
      * {@inheritdoc}
+     *
+     * @param Response $other The guzzle response object
      */
     protected function matches($other)
     {
+        if (!$other instanceof ResponseInterface) {
+            throw new \RuntimeException(sprintf('Expected a GuzzleHttp\Psr7\Response but got %s', get_class($other)));
```

Comment:
i realized we are assuming things here that could lead to fatal errors during the tests.